src/Agent.Worker/StepsRunner.cs

```diff
@@ -47,14 +50,14 @@ public sealed class StepsRunner : AgentService, IStepsRunner
             foreach (IStep step in steps)
             {
                 Trace.Info($"Processing step: DisplayName='{step.DisplayName}', AlwaysRun={step.AlwaysRun}, ContinueOnError={step.ContinueOnError}, Critical={step.Critical}, Enabled={step.Enabled}, Finally={step.Finally}");
+                ArgUtil.Equal(true, step.Enabled, nameof(step.Enabled));
+                ArgUtil.NotNull(step.ExecutionContext, nameof(step.ExecutionContext));
+                ArgUtil.NotNull(step.ExecutionContext.Variables, nameof(step.ExecutionContext.Variables));
 
-                // TODO: Disabled steps may have already been removed. Investigate.
                 // TODO: Run finally even if canceled?
```

Comment:
I verified one of the TODOs. It appears !enabled steps are removed prior to this code executing. I added an assertion that the step is enabled.